chainer/optimizers/adam.py

```diff
@@ -210,6 +277,18 @@ def lr(self):
             DeprecationWarning)
         return self.alpha_t
 
+    @property
+    def bounds(self):
+        if self.t == 0:
+            raise RuntimeError(
+                'Can\'t determine the bounds of AdaBound optimizer '
+                'because the update steps have not been started.')
+        hp = self.hyperparam
+        final_lr = hp.final_lr_rate * hp.alpha
```

Comment:
Did you test this with learning rate decay extensions? (I'm concerned about that this might be different from the original implementation)